src/app/InteractionModelEngine.cpp

```diff
@@ -278,6 +278,37 @@ CHIP_ERROR InteractionModelEngine::ShutdownSubscription(const ScopedNodeId & aPe
     return CHIP_ERROR_KEY_NOT_FOUND;
 }
 
+bool InteractionModelEngine::IsSubjectSubscriptionActive(const FabricIndex & aFabricIndex, const NodeId & subjectID)
+{
+    for (uint16_t index = 0; index < GetNumActiveReadHandlers(); index++)
+    {
+        ReadHandler * readHandler = ActiveHandlerAt(index);
```

Comment:
This is an O(N^2) traversal of the list, no?  Why is this not doing `mReadHandlers.ForEachActiveObject` directly like the other things in this file that want to walk that list?

Reply:
done